test/test_dtype_alu.py

```diff
@@ -160,6 +165,35 @@ def test_float_cast(self, a, dtype): universal_test_cast(a, dtypes.float32, dtyp
   @given(ht.int32, strat.sampled_from(dtypes_float+dtypes_int+dtypes_bool))
   def test_int32_cast(self, a, dtype): universal_test_cast(a, dtypes.int32, dtype)
 
+  @unittest.skipUnless(is_dtype_supported(dtypes.float64, Device.DEFAULT) and is_dtype_supported(dtypes.float16, Device.DEFAULT),
```

Comment:
instead of skip the whole test, just skip the cases when the casting is not possible (metal can do float16 for example but not float64)